### PR TITLE
[SPARK-10541] [Web UI] Allow ApplicationHistoryProviders to provide their own text when there aren't any complete apps

### PR DESCRIPTION
**Pull request by ajbozarth**
## What changes were proposed in this pull request?

I've added a method to `ApplicationHistoryProvider` that returns the html paragraph to display when there are no applications. This allows providers other than `FsHistoryProvider` to determine what is printed. The current hard coded text is now moved into `FsHistoryProvider` since it assumed that's what was being used before.

I chose to make the function return html rather than text because the current text block had inline html in it and it allows a new implementation of `ApplicationHistoryProvider` more versatility. I did not see any security issues with this since injecting html here requires implementing `ApplicationHistoryProvider` and can't be done outside of code.
## How was this patch tested?

Manual testing and dev/run-tests

No visible changes to the UI

### CODE REVIEW

**Comment by ajbozarth** [comment]
Jenkins, retest this please

**Comment by ajbozarth** [comment]
@steveloughran Since you opened the JIRA do you mind taking a look?

**Comment by steveloughran** [comment]
@ajbozarth I'm not a spark committer, I'm not capaclbe of getting stuff in. I did dd one comment to some of the code, otherwise nothing I have issues with. LGTM

**Comment by ajbozarth** [comment]
Thanks @steveloughran, I'm not seeing your code comment though. Also @srowen mind taking a look?

**Comment by steveloughran** [comment]
Oh, I see, new UI had meant I'd left the comment partially incomplete. Sorry. Just the one: printing out the actual log dir location. That makes it much easier to identify a configuration problem and allow anyone with access to the cluster FS just to take a quick ls of the directory to see what's up

File: core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala [hunk 1]
```diff
@@ -262,6 +263,17 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
+  override def getEmptyListingHtml(): Seq[Node] = {
+    <p>
+      Did you specify the correct logging directory? Please verify your setting of
+      <span style="font-style:italic">spark.history.fs.logDirectory</span>
```
Comment (ajbozarth) [inline core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala]: That's a great idea, I'll update it

**Comment by ajbozarth** [comment]
The dir is actually already displayed so I just added a few words to clarify that the current setting is displayed above.

screen shot 2016-10-18 at 3 48 39 pm

**Comment by vanzin** [comment]
LGTM. Merging to master.

**Pull request closed without merging**